index.js

```diff
@@ -0,0 +1,205 @@
+const typeScriptSettings = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    sourceType: 'module',
+    project: './tsconfig.json',
+  },
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
+  ],
+  rules: {
+    '@typescript-eslint/array-type': ['error', { default: 'generic', readonly: 'generic' }],
+    '@typescript-eslint/ban-ts-comment': 'error',
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/explicit-member-accessibility': 'error',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'default',
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+        trailingUnderscore: 'forbid',
+      },
+      {
+        selector: 'typeLike',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE'],
+      },
+      {
+        selector: 'enumMember',
+        format: ['PascalCase'],
+      },
+    ],
+    '@typescript-eslint/no-empty-function': 'error',
+    '@typescript-eslint/no-explicit-any': 'error',
+    '@typescript-eslint/no-inferrable-types': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      { vars: 'all', args: 'after-used', ignoreRestSiblings: true },
+    ],
+    '@typescript-eslint/no-use-before-define': 'error',
+    '@typescript-eslint/prefer-interface': 'off',
+    '@typescript-eslint/prefer-readonly': 'error',
+    camelcase: 'off',
```

Comment:
In a project I have the following rule:

```
        "@typescript-eslint/no-magic-numbers": [
            2,
            {
                "ignoreNumericLiteralTypes": false,
                "ignoreEnums": true,
                "ignore": [-1,0,1],
                "ignoreArrayIndexes": true
            }
        ],
```

Which disallows the use of 'magic' numbers but allows most used `-1`, `0` and `1`. This enforces the dev to name a magic number.

Reply:
This becomes very annoying with animation timings and things like that, right?

Reply:
True that, but in that project we didn't do a lot of animation; you are right tho it can become annoying. If you would scope animations to a file `/* eslint-disable */` on top should suffice.

Reply:
You should never do a `/* eslint-disable */` on top of a file (which is luckily forbidden with an other linting rule), because you disable ALL rule with that statement. If you want to disable something, always specify which rule you want do disable, like `/* eslint-disable @typescript-eslint/no-magic-numbers */`

Reply:
Animation timings are usually values between 0 and 1.5, so we could add these numbers to the ignore list. I tried to following settings in a project:
````    "@typescript-eslint/no-magic-numbers": [
      'error',
      {
        ignoreNumericLiteralTypes: true,
        ignoreEnums: true,
        ignore: [-1, ...(new Array(31).fill(0).map((_, i) => (5 * i) / 100))],
        ignoreArrayIndexes: true,
        ignoreReadonlyClassProperties: true,
      }
    ],
```
And then you filter out the animation timing and end up with only the "real" magic numbers.

One thins: it also complains about this:
```
  public limit: number = 16;
  public pastEventsLimit: number = 6;
```
Which are not really magic numbers in my opinion.

Reply:
Yeah I meant the the latter `/* eslint-disable @typescript-eslint/no-magic-numbers */` was to lazy to copy it all ;). In your last example that should be allowed as they are defined numbers, and not magic.

Reply:
I don't really like this approach of trying to add any number in there that might be used in a context where it warrants its use. Way to random.

IMO linting rules should only be somewhat annoying (i.e. you have to change something in the way you use them, or disable them when valid) when they prevent actual errors. Or when we want to be super strict "on purpose". (or when they can be auto-fixed)

When it's more about "good practices", and they still get in the way a lot, I'd rather have them turned off.

The use of magic numbers is way to arbitrary to get right a lot of the time, so in this case I'd rather focus on education during code reviews and personal judgement.

In your example of a limit of `16`, you could also argue that it should be a CONST at the top of the file or in a shared place. But a linter doesn't understand when it might be useful to do that, and when it doesn't.

Reply:
If we start using the new GSAP we should no longer have magic numbers for timings. So then we could implement this rule